### PR TITLE
Revisit exec env types test

### LINKED ISSUE #1694: Revisit exec env types test
We have old test but we make some changes. Let's revisit and make test coverage for all functions.

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed #1694

### CODE REVIEW

File: chain/x/oracle/types/exec_env_test.go [hunk 1]
```diff
@@ -1,243 +1,173 @@
...
+	"github.com/bandprotocol/bandchain/chain/x/oracle/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+)
+
+func MockExecEnv() *types.ExecEnv {
+	oracleScriptID := types.OracleScriptID(1)
+	calldata := []byte("CALLDATA")
+	validatorAddress1 := sdk.ValAddress([]byte("val1"))
```
Comment (sorawit) [inline chain/x/oracle/types/exec_env_test.go]: Don't use arbitrary val address like this. Please use valid 20 bytes.

File: chain/x/oracle/types/exec_env_test.go [hunk 2]
```diff
@@ -1,243 +1,173 @@
...
+	validatorAddress2 := sdk.ValAddress([]byte("val2"))
+	validatorAddress3 := sdk.ValAddress([]byte("val3"))
+	valAddresses := []sdk.ValAddress{validatorAddress1, validatorAddress2, validatorAddress3}
+	minCount := int64(1)
+	requestHeight := int64(999)
+	requestTime := int64(1581589700)
+	cleintID := "beeb"
+	ibcInfo := types.IBCInfo{SourcePort: "source_port", SourceChannel: "source_channel"}
```
Comment (sorawit) [inline chain/x/oracle/types/exec_env_test.go]: NewIBCInfo

File: chain/x/oracle/types/exec_env_test.go [hunk 3]
```diff
@@ -1,243 +1,173 @@
...
+	require.Equal(t, int64(1581589770), env.GetAggregateBlockTime())
+
+}
+
+func TestGetValidatorAddress(t *testing.T) {
+	env := MockExecEnv()
+
+	valAddrs, err := env.GetValidatorAddress(0)
+	require.NoError(t, err)
+	require.Equal(t, []byte(sdk.ValAddress([]byte("val1"))), valAddrs)
```
Comment (sorawit) [inline chain/x/oracle/types/exec_env_test.go]: ditto

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/oracle/types/exec_env_test.go [hunk 4]
```diff
@@ -1,243 +1,173 @@
...
+	oracleScriptID := types.OracleScriptID(1)
+	calldata := []byte("CALLDATA")
+	validatorAddress1 := sdk.ValAddress([]byte("val1"))
+	validatorAddress2 := sdk.ValAddress([]byte("val2"))
+	validatorAddress3 := sdk.ValAddress([]byte("val3"))
+	valAddresses := []sdk.ValAddress{validatorAddress1, validatorAddress2, validatorAddress3}
+	minCount := int64(1)
+	requestHeight := int64(999)
+	requestTime := int64(1581589700)
+	cleintID := "beeb"
```
Comment (sorawit) [inline chain/x/oracle/types/exec_env_test.go]: typo

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/oracle/types/exec_env_test.go [hunk 5]
```diff
@@ -1,243 +1,195 @@
...
+	requestTime := int64(1581589700)
+	clientID := "beeb"
+	ibcInfo := types.NewIBCInfo("source_port", "source_channel")
+	rawRequestID := []types.ExternalID{1, 2, 3}
+	request := types.NewRequest(oracleScriptID, calldata, valAddresses, minCount, requestHeight, requestTime, clientID, &ibcInfo, rawRequestID)
+	env := types.NewExecEnv(request, int64(1581589770), int64(0))
+	return env
+}
+
+func MockPreparedExecEnv() *types.ExecEnv {
```
Comment (taobun) [inline chain/x/oracle/types/exec_env_test.go]: mockPreparedExecEnv

File: chain/x/oracle/types/exec_env_test.go [hunk 6]
```diff
@@ -1,243 +1,195 @@
...
+	pk3               = ed25519.GenPrivKey().PubKey()
+	addr1             = pk1.Address()
+	addr2             = pk2.Address()
+	addr3             = pk3.Address()
+	validatorAddress1 = sdk.ValAddress(addr1)
+	validatorAddress2 = sdk.ValAddress(addr2)
+	validatorAddress3 = sdk.ValAddress(addr3)
+)
+
+func MockExecEnv() *types.ExecEnv {
```
Comment (taobun) [inline chain/x/oracle/types/exec_env_test.go]: Rename to mockExecutionEnv

File: chain/x/oracle/types/exec_env_test.go [hunk 7]
```diff
@@ -1,243 +1,195 @@
...
+	pk2               = ed25519.GenPrivKey().PubKey()
+	pk3               = ed25519.GenPrivKey().PubKey()
+	addr1             = pk1.Address()
+	addr2             = pk2.Address()
+	addr3             = pk3.Address()
+	validatorAddress1 = sdk.ValAddress(addr1)
+	validatorAddress2 = sdk.ValAddress(addr2)
+	validatorAddress3 = sdk.ValAddress(addr3)
+)
+
```
Comment (taobun) [inline chain/x/oracle/types/exec_env_test.go:22]: Add mockFreshEnv using to prepare test

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/oracle/types/exec_env_test.go (hunk 7 above)
Comment (taobun) [reply chain/x/oracle/types/exec_env_test.go:22]: For test RequestExternalData functions

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by sorawit — DISMISSED** [review]

File: chain/x/oracle/types/exec_env_test.go [hunk 8]
```diff
@@ -1,243 +1,226 @@
...
-
-// // 	// Get report from invalid validator index
-// // 	_, _, envErr = env.GetExternalData(42, 2)
-// // 	require.NotNil(t, envErr, "validator out of range")
-
-// // 	// Get report from invalid validator index
-// // 	_, _, envErr = env.GetExternalData(42, -2)
-// // 	require.NotNil(t, envErr, "validator out of range")
-// // }
+package types_test
```
Comment (taobun) [inline chain/x/oracle/types/exec_env_test.go]: I think we can use package types

File: chain/x/oracle/types/exec_env_test.go [hunk 9]
```diff
@@ -1,243 +1,226 @@
...
+
+import (
+	"testing"
+
+	"github.com/bandprotocol/bandchain/chain/x/oracle/types"
+	"github.com/tendermint/tendermint/crypto/ed25519"
+
```
Comment (taobun) [inline chain/x/oracle/types/exec_env_test.go]: Remove endline and bring our import to bottom.

**Review by taobun — CHANGES_REQUESTED** [review]

**Comment by Benzbeeb** [comment]
I’m closing this pull request because of outdated code. Now we have go-owasm.([link](https://github.com/bandprotocol/bandchain/pull/1858))

**Pull request closed without merging**